addOns/dns/src/main/java/org/zaproxy/addon/dns/ExtensionDns.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.dns;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+
+/** An ZAP extension which performs DNS operations to get information or vulnerabilities. */
+public class ExtensionDns extends ExtensionAdaptor {
```

Comment:
Should override `getUIName()`.

Reply:
Why? 
Default implementation calls to `getName()`, what returns `name`. So overriding it to `return name`  won't help, right?

Reply:
It should return an i18n name, e.g.: https://github.com/zaproxy/zap-extensions/blob/f4fc5f6f4a12a3b3170bf987dce1ea4911558816/addOns/kotlin/src/main/java/org/zaproxy/addon/kotlin/ExtensionKotlin.java#L58-L61

Reply:
It should be a user friendly i18n string because it’s shown in the options dialog.